### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by rodrigozoff**
Hi! Anothers improvements like:

- Grouped operations by Tags
- Api Factory
- Service factory (see readme)
- Naming convention for : model name, model & enum properties name, api name, service name. Support camel, pascal, snake, snake-upper.  (see readme)

Thanks!

### CODE REVIEW

**Comment by hosseinmd** [comment]
Please don't put everything in a PR, I don't understand what's going on.

**Comment by rodrigozoff** [comment]
Sorry, yes I understand. For the next PR, I will improve this.

**Comment by tbowmo** [comment]
I would also prefer splitting this PR into multiple smaller PRs, helps focus on a single issue when reviewing the changes.

Create 1 PR for each bulletpoint in your list.

https://www.thedroidsonroids.com/blog/splitting-pull-request

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (tbowmo) [inline .vscode/launch.json]: Perhaps we should add .vscode to .gitignore file, as it can contain developer specific settings..

Comment (rodrigozoff) [reply .vscode/launch.json]: Okey, I added to .gitignore.
Do you prefer that we close this pull request and that I make several small pull requests?

Comment (rodrigozoff) [reply .vscode/launch.json]: Do you prefer that we close this pull request and that I make two small pull requests?
- adding naming conventions
- adding tags & service factory & api factory & naming for tags

**Comment by rodrigozoff** [comment]
Do you prefer that we close this pull request and that I make two small pull requests?
- adding naming conventions
- adding tags & service factory & api factory & naming for tags

**Comment by hosseinmd** [comment]
I close this PR, please do not feel bad. This is not possible to merge.

**Pull request closed without merging**